extensions/wikia/BrokenRenameFix/SpecialBrokenRenameFixController.class.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Class SpecialBrokenRenameFixController
+ *
+ * A special page with an interface to rerun local rename jobs.
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ * @copyright (c) 2015 Wikia, Inc.
+ * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ * @package BrokenRenameFix
+ */
+
+class SpecialBrokenRenameFixController extends WikiaSpecialPageController {
+
+	const BRF_FORM_SUBMIT = 'submit';
+
+	private
+		$notices = [],
+		$userId,
+		$oldName,
+		$newName;
+
+	function __construct() {
+		parent::__construct( 'BrokenRenameFix', 'renameuser', true );
+	}
+
+	public function index() {
+		if ( !$this->getContext()->getUser()->isAllowed( 'renameuser' ) ) {
+			throw new PermissionsError( 'renameuser' );
+		}
+
+		Wikia::addAssetsToOutput( 'special_broken_rename_fix' );
+
+		$this->wg->Out->setPageTitle( wfMessage( 'brf-title' )->escaped() );
+
+		$this->setMessages();
+		$this->setFormData();
+
+		if ( $this->getPar() === self::BRF_FORM_SUBMIT ) {
+			$this->submit();
+		}
+
+		if ( !empty( $this->notices ) ) {
+			$this->formNotices = implode( "\n", $this->notices );
+		}
+	}
+
+	/**
+	 * Creates a prioritized task that reruns the local rename jobs.
+	 */
+	private function submit() {
+		if ( $this->validateForm() ) {
+			/**
+			 * Prevent sending the data again on refresh.
+			 */
+			unset( $_POST );
+
+
+			$task = new BrokenRenameFixTask();
+			$task->call( 'rerunRenameScript', $this->userId, $this->oldName, $this->newName );
+			$task->prioritize();
+			$task->queue();
+
+			$this->addNotice( wfMessage( 'brf-success' )->escaped() . $this->getLogsLink(), 'success' );
+		} else {
+			$this->prefillFormValues();
+		}
+	}
+
+	/**
+	 * Validates the form's input and returns a status of the checks. It also adds notices
+	 * on each step if something goes wrong.
+	 *
+	 * @return bool
+	 */
+	private function validateForm() {
+		$status = true;
+
+		$editToken = $this->request->getVal( 'brf-form-token' );
+		if ( !$this->request->wasPosted() || !$this->wg->User->matchEditToken( $editToken ) ) {
+			$this->addNotice( wfMessage( 'brf-error-invalid-request' )->escaped() );
+			$status = false;
+		}
+
+		$this->userId = $this->request->getVal( 'brf-form-user-id' );
+		$this->oldName = $this->request->getVal( 'brf-form-old-name' );
+		$this->newName = $this->request->getVal( 'brf-form-new-name' );
+
+		if ( empty( $this->userId ) || empty( $this->oldName ) || empty( $this->newName ) ) {
+			$this->addNotice( wfMessage( 'brf-error-empty-fields' )->escaped() );
+			$status = false;
+		}
+
+		$usernameFromId = User::newFromId( $this->userId )->getName();
+		if ( $usernameFromId !== $this->oldName && $usernameFromId !== $this->newName ) {
```

Comment:
For rerunning local renames, the user ID must match the new name, not the old name. If the user ID is still associated with the old user name, then the global rename needs to be rerun, not just the local.